LPS-83764 3

dantewang opened the pull request.

Commit:
LPS-83764 add unit test for LiferayObjectWrapper

dantewang committed:
LPS-83764 Enumeration is also generic

dantewang committed:
LPS-83764 Let TestLiferay* extend with generic

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

dantewang committed:
LPS-83764 Rename variables after renaming inner classes

dantewang commented:
@tinatian 
1. Changed `Throwable` to `Exception` and now it can be fully covered.
1. Applied generic types. In `testHandleUnknownType`, `?` is used as the type, is that OK?
2. Renamed inner test classes to TestLiferay*.
3. SF does pass without the message in `Assert.fail()`, but we added one.

cc @caixiangyue

dantewang commented:
ci:test:relevant

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:relevant - 19 out of 19 jobs passed in 58 minutes 2 seconds 429 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/992557623a0840100d9e5122234ba5eac8ee2e72">992557623a0840100d9e5122234ba5eac8ee2e72</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/dc7622217d51258ba20c221af52a243e6f9d2f84">dc7622217d51258ba20c221af52a243e6f9d2f84</a></p>19 out of 19 jobs PASSED<details><summary><strong>19 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest(master)/3192/">test-portal-acceptance-pullrequest(master)</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/244680/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/222294/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/213493/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/168465/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/240166/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/213495/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/160190/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/213494/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/168464/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/244681/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/240167/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/240277/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/222295/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/160191/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/249581/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/97047/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/240276/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/1618/">test-portal-acceptance-pullrequest-dist(master)/dist-jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-14.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/3192/jenkins-report.html">here</a>.</h5></details></html>

tinatian commented:
@dantewang @caixiangyue 
Let's do more in the test than just making it 100% covered. We should focus on the feature we want to test. You should understand what this class is for. It is to wrap and java object to a type which freemarker can read.
For example, you are doing this:

             TemplateNode templateNode = new TemplateNode(null, "", "", "", null);

		TemplateModel result = liferayObjectWrapper.wrap(templateNode);

		Assert.assertTrue(result instanceof LiferayTemplateModel);`
I would like to test it like this:

              TemplateNode templateNode = new TemplateNode(
			null, "testName", "", "", null);

		TemplateModel result = liferayObjectWrapper.wrap(templateNode);

		Assert.assertTrue(result instanceof LiferayTemplateModel);
		
		LiferayTemplateModel liferayTemplateModel = 
			(LiferayTemplateModel)result;
		
		Assert.assertEquals(
			"testName", liferayTemplateModel.get("name").toString());`

This is just an example to show you what I want, please refine the code and resend.


Thanks.
Tina.
